### PR TITLE
Spatial_Engine: Query Thickness of IProfiles

### LINKED ISSUE #2171: Spatial_Engine: Query IProfile Thicknesses
#### Description:

Helpful query for any method needing easy access to this parameter (i.e. building toRevit converts for linear MEP elements where `Engine.Create.SectionProfile` utilises these thickness parameters)

### PR DESCRIPTION
**Pull request by michaelhoehn**
### Issues addressed by this PR

Closes #2171 

### Test files

[Here](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM_Engine/Spatial_Engine/Issues/Spatial_Engine-%232171-IProfileThicknessQuery/IProfileThickness.gh?csf=1&web=1&e=ozLBf5)

### CODE REVIEW

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
For this to feel a bit more useful from the side of the UI, I think it should be expanded to cover all profiles, and/or needs a fallback method!

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Definitely needs a fall back method echoing @IsakNaslundBh regardless - but would also agree it would be good to cover all profiles at this point.

**Comment by michaelhoehn** [comment]
Thanks @IsakNaslundBh and @FraserGreenroyd. I've added support for each ShapeProfile that has a Thickness property. This particular method would really only support three types: Box, Tube and Kite. The fallback method error message should reflect which profile types aren't supported. 

I prototyped a multi-output method for the others but felt that may constitute a separate method to cover multiple thicknesses.

File: Spatial_Engine/Query/Thickness.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+namespace BH.Engine.Spatial
+{
+    public static partial class Query
+    {
+        /******************************************/
+        /****          ShapeProfiles           ****/
+        /******************************************/
+
+        [Description("Returns the thickness of a boxProfile.")]
+        [Input("boxProfile", "The boxProfile to query.")]
```
Comment (FraserGreenroyd) [inline Spatial_Engine/Query/Thickness.cs:37]:
```suggestion
        [Input("boxProfile", "The BoxProfile to query.")]
```

File: Spatial_Engine/Query/Thickness.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+namespace BH.Engine.Spatial
+{
+    public static partial class Query
+    {
+        /******************************************/
+        /****          ShapeProfiles           ****/
+        /******************************************/
+
+        [Description("Returns the thickness of a boxProfile.")]
```
Comment (FraserGreenroyd) [inline Spatial_Engine/Query/Thickness.cs:36]:
```suggestion
        [Description("Returns the thickness of a BoxProfile.")]
```

File: Spatial_Engine/Query/Thickness.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        [Input("boxProfile", "The boxProfile to query.")]
+        [Output("thickness", "Thickness of the boxProfile.", typeof(Length))]
```
Comment (FraserGreenroyd) [inline Spatial_Engine/Query/Thickness.cs:38]:
```suggestion
        [Output("thickness", "Thickness of the BoxProfile.", typeof(Length))]
```

File: Spatial_Engine/Query/Thickness.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        public static double Thickness(this BoxProfile boxProfile)
+        {
+            return boxProfile.Thickness;
+        }
+
+        /******************************************/
+
+        [Description("Returns the thickness of a kiteProfile.")]
+        [Input("kiteProfile", "The kiteProfile to query.")]
+        [Output("thickness", "Thickness of the kiteProfile.", typeof(Length))]
```
Comment (FraserGreenroyd) [inline Spatial_Engine/Query/Thickness.cs:48]:
```suggestion
        [Description("Returns the thickness of a KiteProfile.")]
        [Input("kiteProfile", "The KiteProfile to query.")]
        [Output("thickness", "Thickness of the KiteProfile.", typeof(Length))]
```

File: Spatial_Engine/Query/Thickness.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        public static double Thickness(this KiteProfile kiteProfile)
+        {
+            return kiteProfile.Thickness;
+        }
+
+        /******************************************/
+
+        [Description("Returns the thickness of a tubeProfile.")]
+        [Input("tubeProfile", "The tubeProfile to query.")]
+        [Output("thickness", "Thickness of the tubeProfile.", typeof(Length))]
```
Comment (FraserGreenroyd) [inline Spatial_Engine/Query/Thickness.cs:58]:
```suggestion
        [Description("Returns the thickness of a TubeProfile.")]
        [Input("tubeProfile", "The TubeProfile to query.")]
        [Output("thickness", "Thickness of the TubeProfile.", typeof(Length))]
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Documentation could do with improving entirely but as a starting point it's ok - just needs some grammar changes, see suggestions.

Otherwise I feel this is appropriate following our conversation where the other Profiles have multiple thickness attributes, and choosing which one to show may be misleading, or summing them may also be misleading. @IsakNaslundBh what would be your thoughts on that given you implemented most of the profile I suspect?

File: Spatial_Engine/Query/Thickness.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Spatial_Engine/Query/Thickness.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright compliance

**Review by FraserGreenroyd — APPROVED** [review]
This LGTM now, thanks @michaelhoehn

**Pull request merged**